synapse/lib/ast.py

```diff
@@ -4641,55 +4661,78 @@ async def run(self, runt, genr):
-                        if subn.form.isrunt:
-                            mesg = f'Edges cannot be used with runt nodes: {subn.form.full}'
-                            raise self.addExcInfo(s_exc.IsRuntForm(mesg=mesg, form=subn.form.full))
-                        await subn.delEdge(verb, iden)
-
-                else:
-                    async with node.snap.getEditor() as editor:
-                        proto = editor.loadNode(node)
+            if not constverb:
+                verb = await tostr(await self.kids[0].compute(runt, path))
+                allowed(verb)
+
+            if isvar:
+                valu = await vkid.compute(runt, path)
+                async with contextlib.aclosing(self.yieldFromValu(runt, valu, vkid)) as agen:
+                    if self.n2:
+                        iden = node.iden()
+                        async for subn in agen:
+                            await subn.delEdge(verb, iden, extra=self.addExcInfo)
+                    else:
+                        async with node.snap.getEditor() as editor:
+                            proto = editor.loadNode(node)
+                            async for subn in agen:
+                                await self.delEdgeN1(editor, proto, verb, subn)
 
+            else:
+                async with runt.getSubRuntime(query) as subr:
+                    if self.n2:
+                        iden = node.iden()
                         async for subn, subp in subr.execute():
-                            if subn.form.isrunt:
-                                mesg = f'Edges cannot be used with runt nodes: {subn.form.full}'
-                                raise self.addExcInfo(s_exc.IsRuntForm(mesg=mesg, form=subn.form.full))
+                            await subn.delEdge(verb, iden, extra=self.addExcInfo)
+                    else:
+                        async with node.snap.getEditor() as editor:
+                            proto = editor.loadNode(node)
+                            async for subn, subp in subr.execute():
+                                await self.delEdgeN1(editor, proto, verb, subn)
 
-                            await proto.delEdge(verb, subn.iden())
-                            await asyncio.sleep(0)
+            yield node, path
 
-                            if len(proto.edgedels) >= 1000:
-                                nodeedits = editor.getNodeEdits()
-                                if nodeedits:
-                                    await node.snap.applyNodeEdits(nodeedits)
-                                proto.edgedels.clear()
+    async def delEdgeN1(self, editor, proto, verb, subn):
+        if subn.form.isrunt:
+            mesg = f'Edges cannot be used with runt nodes: {subn.form.full}'
+            raise self.addExcInfo(s_exc.IsRuntForm(mesg=mesg, form=subn.form.full))
 
-            yield node, path
+        await proto.delEdge(verb, subn.iden())
+        await asyncio.sleep(0)
+
+        if len(proto.edgedels) >= 1000:
```

Comment:
Should this type of behavior be in the `proto` to automatically flush node edits?

Reply:
Yeah would probably be good to move it there